### ISSUE TITLE
Input could not be saved

### ISSUE THREAD

**Issue by veddegre** [Cisco-AMP/amp4e_splunk_events_input#6]
I am running Splunk 7.1 and have installed Cisco Amp for Endpoints Events Input (1.1.0) on a heavy forwarder. I can setup the configuration OK (api.amp.cisco.com / Client ID / API Key). The API has read/write access. The index is setup and accessible from the heavy forwarder (forwarding to a 3 member index cluster). When I go to create an input and save it I get the following error message:

**Input could not be saved:**
the server responded with status 401

I have verified network connectivity and made the changes / tests as outlined in this issue: https://github.com/Cisco-AMP/amp4e_splunk_events_input/issues/1 

I would greatly appreciate any suggestions as to how I can solve this issue.

**Comment by nichus** [comment]
I'm actually seeing the same issue, under the same conditions.  Splunk v7.1.1, TA v1.1.0

My attempts to create a stream manually result in the following error:

"errors": [
   {"error_code": 400,"description": "Bad Request","details": ["the server responded with status 401"]}
]

**Comment by samsonnguyen** [comment]
does your AMP for Endpoints API key have read and write privileges?

**Comment by samsonnguyen** [comment]
a 401 is indicative that that the API key is not correct or that you have the incorrect api host field. Additionally, private cloud instances of AMP4E is not supported.

We host AMP for Endpoints across three regions, please ensure that the api host field (during configuration) is correct

| Region | api host |
|---| ---|
|North America | api.amp.cisco.com |
|Europe | api.eu.amp.cisco.com |
| APJC | api.apjc.amp.cisco.com |

**Comment by nichus** [comment]
I'm no longer supporting the customer that I was at the time I added my comments so I can't add too much more to this conversation, but I do know that when we tried again the next day, with the settings unchanged, the 401 error went away. So, while I agree that an http 401 error does indicate an authentication failure, this time my guess was that this was a transient server/network/unknown issue of some kind.

As far as I'm aware at this point, my customer is no longer receiving the 401 error.

Hope that helps.
